### ISSUE TITLE
Kombu version conflict

### ISSUE THREAD

**Issue by asyncee** [nameko/nameko-amqp-retry#23]
Hello, we have nameko 2.11.0 which requires kombu > 4. We want to use nameko-amqp-retry, but it requires kombu < 4.

The following error is displayed in logs:

```
nameko-amqp-retry 0.6.0 has requirement kombu<4,>=3.0.25, but you'll have kombu 4.2.1 which is incompatible.
```

**Comment by asyncee** [comment]
I got it working by changing one line (141) in file backoff.py:

from

```
maybe_declare(queue, conn, retry=True, **DEFAULT_RETRY_POLICY)
```

to

```
maybe_declare(queue, producer.channel, retry=True, **DEFAULT_RETRY_POLICY)
```

**Update**:

After backoff has reached it's limit, exception is raised:
```
amqp.exceptions.ChannelError: Basic.return: (312) NO_ROUTE
```

**Comment by cblegare** [comment]
Hi there,

I tried to fix it myself but got stuck trying to run the tests with the existing code.  I think there are a few compatibility issues, meaning I don't think test would pass for any nameko>2.7.

Sadly, at least for now, these problems deterred me from working on this issue

**Comment by mattbennett** [comment]
Fixed by #26
